### ISSUE TITLE
Issue with external scripts when in dev mode

### ISSUE THREAD

**Issue by nikitaWebApps** [nuxt/nuxt#29123]
### Environment

- Operating System: `Darwin`
- Node Version:     `v22.7.0`
- Nuxt Version:     `3.13.2`
- CLI Version:      `3.13.2`
- Nitro Version:    `2.9.7`
- Package Manager:  `npm@10.8.2`
- Builder:          `vite`
- User Config:      `modules`, `css`, `compatibilityDate`
- Runtime Modules:  `@nuxtjs/tailwindcss@6.12.1`, `@nuxt/image@1.8.0`, `@nuxt/content@2.13.2`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-1bvrus?file=app.vue

### Describe the bug

Hello!

I'm trying to add a CRM form that is being provided via external scripts. When running in dev mode, the following error appears in the console:

TypeError: Cannot read properties of undefined (reading 'extend')

and the form won't render.

[Here’s a minimal reproduction link.](https://stackblitz.com/edit/github-1bvrus?file=app.vue)

The issue doesn’t occur when I write the same code using pure Vue or in a Vue SSR sandbox. [Here’s the expected result using pure Vue SSR.](https://stackblitz.com/edit/github-gxnq1d?file=src%2FApp.vue)
It also doesn’t appear after I run npx nuxi generate.

My guess is that it’s somehow connected to the Nuxt dev server, but unfortunately, I don’t have enough expertise yet to understand how everything works under the hood.

I’d appreciate any suggestions!

**Comment by ndragun92** [comment]
Looks like this script
![image](https://github.com/user-attachments/assets/d469127d-b3ee-406d-8ecd-1563ff4acdbf)

Doesnt have access to Vue instance and therefor it fails to execute code

**Comment by nikitaWebApps** [comment]
Yes, but as this form is generated by SaaS, there’s only so much I can do. It’s also supposed to be separate from my site’s code and should work in most situations.

My biggest concerns with this issue are:

Why does it work in pure Vue then?
Why doesn’t it work after I generate a static page in Nuxt?

It seems like there might be differences in how the page is rendered in Vue vs Nuxt (even though Nuxt uses Vue), even with SSR turned off. That’s what I’m trying to understand) May be there's a bug or a room for improvement for Nuxt in this case.

**Comment by harlan-zw** [comment]
The package is assuming that Vue 2 is available. Nuxt uses Vue 3. I'd recommend getting in contact with the SaaS and getting them to bundle their dependencies instead of assuming compatibility. Better yet, have them make a Nuxt module :)
